src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -125,6 +125,8 @@ internal void GetDeclarationDiagnostics(DiagnosticBag addTo)
             GetAttributes();
             GetReturnTypeAttributes();
 
+            PostDecodeAllMethodAttributes(_declarationDiagnostics);
```

Comment:
>PostDecodeAllMethodAttributes(_declarationDiagnostics); [](start = 12, length = 55)

Should consolidate with other async-related checks performed for local functions and probably share code with regular methods so that there was one, well defined place where we perform async-related checks.